classes/01class/exercises/c01-git03/zarajoy/c01-git03.md

```diff
@@ -0,0 +1,16 @@
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+Yes. The repo size was more than what I expected after I added the big file (eg increase in size was more than the size of the file i added). 
+
+2. What is the reason for this problem?
+A copy of the file was stored on the history on the folder /.git/objects with the 'git add file.txt; git commit -m "Add big file"' command
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file" HEAD
+Source :https://myopswork.com/how-remove-files-completely-from-git-repository-history-47ed3e0c4c35
+
+4. What are the consequences of the previous action?
+you cannot reverse the removal. you should probably make a back up of the repo somewhere.
```

Comment:
Other users still have older histories, so they could branch out from there and "save" the work.

But what would they notice and trying to push their changes?

Reply:
from what i have read filter-branch is a poor choice especially if other users with their own clones are involved. for a singular local instance it is probably ok?
filter-branch can cause multiple copies of each commit, some with removed information, others without it. 
filter-repo might be better for public repos.

Reply:
Can you add that extra information to your answer? :)